plugin/src/main/java/io/jenkins/plugins/casc/TokenReloadAction.java

```diff
@@ -0,0 +1,72 @@
+package io.jenkins.plugins.casc;
+
+import com.google.common.base.Strings;
+import hudson.Extension;
+import hudson.model.UnprotectedRootAction;
+import java.io.IOException;
+import java.util.logging.Logger;
+import javax.annotation.CheckForNull;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.httpclient.HttpStatus;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+@Extension
+public class TokenReloadAction implements UnprotectedRootAction {
+    public static final Logger LOGGER = Logger.getLogger(TokenReloadAction.class.getName());
+
+    public static final String URL_NAME = "reload-configuration-as-code";
+    public static final String RELOAD_TOKEN_PROPERTY = "casc.reload.token";
+    public static final String RELOAD_TOKEN_QUERY_PARAMETER = "casc-reload-token";
+
+    @CheckForNull
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @CheckForNull
+    @Override
+    public String getDisplayName() {
+        return "Reload Configuration as Code";
+    }
+
+    @CheckForNull
+    @Override
+    public String getUrlName() {
+        return URL_NAME;
+    }
+
+    @RequirePOST
+    public void doIndex(StaplerRequest request, StaplerResponse response) throws IOException {
+        String token = getReloadTokenProperty();
+
+        if (Strings.isNullOrEmpty(token)) {
+            LOGGER.warning("Configuration reload via token is not enabled");
```

Comment:
hmm I think we should return an error here, returning 200 isn't ideal, 
404 i guess with a  message unless anyone has any better ideas?

Reply:
done